api4/image.go

```diff
@@ -6,16 +6,26 @@ package api4
 import (
 	"net/http"
 	"net/url"
+	"regexp"
 
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+var protocolRelativeURLRegex = regexp.MustCompile(`^(\\|\/){2}.+`) // can be any one of //, \\, /\, or \/
+
 func (api *API) InitImage() {
 	api.BaseRoutes.Image.Handle("", api.ApiSessionRequiredTrustRequester(getImage)).Methods("GET")
 }
 
 func getImage(c *Context, w http.ResponseWriter, r *http.Request) {
 	actualURL := r.URL.Query().Get("url")
+	if protocolRelativeURLRegex.MatchString(actualURL) {
```

Comment:
A better way is to let the `url.Parse` happen, and then just check for no scheme. That's what I plan to do for the other ticket. Unless you already tried this and found some issue?

Reply:
Hmm, it didn't work well with other than `//` prefixes. If I understand correctly you mean something like [this](https://play.golang.org/p/e1fxedA0aEo) right?

Reply:
Yep, right. This is what I am doing for image_proxy

```go
	if imageURL == "" || proxy.siteURL == nil {
		return imageURL
	}
	// Parse url, bypass if error in parsing.
	parsedURL, err := url.Parse(imageURL)
	if err != nil {
		return imageURL
	}
	// If host is same as siteURL host, return.
	if parsedURL.Host == proxy.siteURL.Host || parsedURL.Host == "" {
		return imageURL
	}
	// Handle protocol-relative URLs.
	if parsedURL.Scheme == "" {
		parsedURL.Scheme = proxy.siteURL.Scheme
	}
```

Reply:
Oh, I see. I was trying to fix the occasions where the URL can start with `\\`, `\/` and `/\`. I think we can defer it to the browser so that to be fixed.